dot/services_test.go

```diff
@@ -0,0 +1,656 @@
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/rpc"
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/sync"
+	"github.com/ChainSafe/gossamer/dot/system"
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/life"
+	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_createBlockVerifier(t *testing.T) {
+	cfg := NewTestConfig(t)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := nodeInstance.createStateService(cfg)
+	require.NoError(t, err)
+
+	stateSrvc.Block = &state.BlockState{}
+	stateSrvc.Epoch = &state.EpochState{}
+
+	tests := []struct {
+		name      string
+		service   *state.Service
+		expectNil bool
+		err       error
+	}{
+		{
+			name:      "nil BlockState test",
+			service:   &state.Service{},
+			expectNil: true,
+			err:       errors.New("cannot have nil EpochState"),
+		},
+		{
+			name:      "working example",
+			service:   stateSrvc,
+			expectNil: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nodeInstance.createBlockVerifier(tt.service)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.expectNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func Test_createRuntimeStorage(t *testing.T) {
+	cfg := NewTestConfig(t)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := nodeInstance.createStateService(cfg)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name           string
+		service        *state.Service
+		expectedBaseDB *state.BaseState
+		err            error
+	}{
+		{
+			name:           "working example",
+			service:        stateSrvc,
+			expectedBaseDB: stateSrvc.Base,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nodeInstance.createRuntimeStorage(tt.service)
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, tt.expectedBaseDB, got.BaseDB)
+			assert.NotNil(t, got.LocalStorage)
+			assert.NotNil(t, got.PersistentStorage)
+			assert.IsType(t, &runtime.NodeStorage{}, got)
```

Comment:
unneeded since it returns a struct pointer, not an interface, so it's checked at compile time already

```suggestion
```

Reply:
removed.